src/fm/test/check.clj

```diff
@@ -1,72 +1,63 @@
 (ns fm.test.check
-  (:require [clojure.alpha.spec :as s]
-            [clojure.alpha.spec.test :as stest]
-            [fm.macros :refer [defm]]
-            [fm.utils :as fm]
-            [fm.test.check :as check]))
+  (:require
+   [clojure.alpha.spec :as s]
+   [clojure.alpha.spec.test :as stest]
+   [fm.utils :as fm]))
 
-(s/def ::namespaces
-  (s/coll-of symbol?
-             :distinct true))
-
-(s/def ::check-result-pass
+(s/def :clojure.spec.test.check/ret-pass
   (s/schema
-   {:num-tests       nat-int?
-    :pass?           boolean?
+   {:pass?           true?
+    :num-tests       nat-int?
     :result          any?
     :seed            nat-int?
     :time-elapsed-ms nat-int?}))
 
-(s/def ::check-result-fail
+(s/def :clojure.spec.test.check/ret-fail
   (s/schema
-   {:fail            any?
-    :failed-after-ms nat-int?
-    :failing-size    any?
+   {:pass?           false?
     :num-tests       nat-int?
-    :pass?           boolean?
     :result          any?
-    :result-data     any?
     :seed            nat-int?
+    :failed-after-ms nat-int?
+    :fail            any?
+    :failing-size    any?
+    :result-data     any?
     :shrunk          any?}))
 
-(s/def ::check-result
+(s/def :clojure.spec.test.check/ret
+  (s/or
+   :pass :clojure.spec.test.check/ret-pass
+   :fail :clojure.spec.test.check/ret-fail))
+
+(s/def :clojure.spec.test.check/result
   (s/schema
-   {:clojure.spec.test.check/ret (s/or
-                                  ::check-result-pass
-                                  ::check-result-fail)
-    :spec                        s/spec?
-    :sym                         symbol?}))
+   [:clojure.spec.test.check/ret
+    {:spec s/spec?
+     :sym  symbol?}]))
 
-(defn fmdef!
-  "Takes a defm and passes it through to s/fdef."
+(defn meta->fdef!
   [{:keys [fm/sym fm/args]}]
   (eval
    `(s/fdef ~sym
       :args ~(s/form args)
-      :ret fm/not-anomaly?)))
+      :ret  fm/not-anomaly?)))
 
-(defn ns->fnsym
-  "Extracts all defms from the given namespace."
+(defn fdef-ns-fms!
+  "lazily fdef all fms in the given namespace."
   [ns-sym]
-  (->> ns-sym ns-publics vals (map deref)))
-
-(defn fms-from-ns
-  "Build a list of fm meta for all fms in a namespace."
-  [namespaces]
   (->>
-   namespaces
-   (mapcat ns->fnsym)
+   (ns-publics ns-sym)
+   (vals)
+   (map deref)
    (filter fm/fm?)
-   (map (comp fmdef! meta))))
+   (map (comp meta->fdef! meta))))
 
-(defm check
-  ^{:fm/doc
-    "Iterate through namespaces, extracting any fns that have fm metadata,
-registering those with s/fdef, and finally running them against s/check."
-    :fm/args ::namespaces
-    :fm/ret  (s/coll-of ::check-result)}
-  [namespaces]
+(defn check!
+  [ns-sym-or-syms]
   (->>
-   namespaces
-   (fms-from-ns)
+   (if (vector? ns-sym-or-syms)
+     ns-sym-or-syms
+     (vector ns-sym-or-syms))
+   (mapcat fdef-ns-fms!)
+   (into [])
```

Comment:
This forces the realization of the `mapcat` result, which in turn forces the calls to `fdef` to be evaluated. I think because `stest/check` is a macro, it wasn't evaluating its arguments, so the results of `mapcat` weren't being realized, and `stest/check` was more than happy to go to work on an empty (i.e. unrealized) sequence.

If `mapcat` weren't lazy (e.g. like `mapv`), this wouldn't have been an issue. I suppose this arguably should have been
```
(->>
  ,,,
  (into [] (mapcat fdef-ns-fms!))
  ,,,)
```
because `mapcat` produces a transducer, but we can tweak that in a future PR.

Reply:
This might also be entirely misguided, but at least it works as intended.

Reply:
If it ain't broke.... 👍